Rubberduck.CodeAnalysis/Inspections/Concrete/FunctionReturnValueNeverUsedInspection.cs

```diff
@@ -0,0 +1,311 @@
+        {
+            return declaration.References
+                .Where(IsProcedureCallReference);
+        }
+
+        private static bool IsProcedureCallReference(IdentifierReference reference)
+        {
+            return reference?.Declaration != null
+                   && !reference.IsAssignment
+                   && !reference.IsArrayAccess
+                   && !reference.IsInnerRecursiveDefaultMemberAccess
+                   && IsCalledAsProcedure(reference.Context);
+        }
+
+        private static bool IsCalledAsProcedure(ParserRuleContext context)
+        {
+            var callStmt = context.GetAncestor<VBAParser.CallStmtContext>();
+            if (callStmt == null)
+            {
+                return false;
+            }
+
+            //If we are in an argument list, the value is used somewhere in defining the argument.
+            var argumentListParent = context.GetAncestor<VBAParser.ArgumentListContext>();
+            if (argumentListParent != null)
+            {
+                return false;
+            }
+
+            //Member accesses are parsed right-to-left, e.g. 'foo.Bar' is the parent of 'foo'.
+            //Thus, having a member access parent means that the return value is used somehow.
+            var ownFunctionCallExpression = context.Parent is VBAParser.MemberAccessExprContext methodCall
+                ? methodCall
+                : context;
+            var memberAccessParent = ownFunctionCallExpression.GetAncestor<VBAParser.MemberAccessExprContext>();
+            if (memberAccessParent != null)
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        protected override string ResultDescription(Declaration declaration)
+        {
+            var functionName = declaration.QualifiedName.ToString();
+            return string.Format(InspectionResults.FunctionReturnValueNeverUsedInspection, functionName);
+        }
+
+        //protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        //{
+        //    var interfaceMembers = State.DeclarationFinder.FindAllInterfaceMembers().ToList();
+        //    var interfaceImplementationMembers = State.DeclarationFinder.FindAllInterfaceImplementingMembers();
+        //    var functions = State.DeclarationFinder
+        //        .UserDeclarations(DeclarationType.Function)
+        //        .Where(item => item.References.Any(r => !IsReturnStatement(item, r) && !r.IsAssignment))
+        //        .ToList();
+        //    var interfaceMemberIssues = GetInterfaceMemberIssues(interfaceMembers);
+        //    var nonInterfaceFunctions = functions.Except(interfaceMembers.Union(interfaceImplementationMembers));
+        //    var nonInterfaceIssues = GetNonInterfaceIssues(nonInterfaceFunctions);
+        //    return interfaceMemberIssues.Union(nonInterfaceIssues);
+        //}
```

Comment:
Should this be removed?

Reply:
Oh, forgot to remove that after testing that things still work. Will remove.